chain/actors/actor_storagepower.go

```diff
@@ -426,6 +549,87 @@ func pledgeCollateralForSize(vmctx types.VMContext, size, totalStorage types.Big
 	return types.BigAdd(powerCollateral, perCapCollateral), nil
 }
 
+func (spa StoragePowerActor) CheckProofSubmissions(act *types.Actor, vmctx types.VMContext, param *struct{}) ([]byte, ActorError) {
+	if vmctx.Message().From != StoragePowerAddress {
+		return nil, aerrors.New(1, "CheckProofSubmissions is only callable as a part of tipset state computation")
+	}
+
+	bucketID := vmctx.BlockHeight() % build.ProvingPeriodDuration
+
+	var self StoragePowerState
+	old := vmctx.Storage().GetHead()
+	if err := vmctx.Storage().Get(old, &self); err != nil {
+		return nil, err
+	}
+
+	buckets, eerr := amt.LoadAMT(types.WrapStorage(vmctx.Storage()), self.ProvingBuckets)
+	if eerr != nil {
+		return nil, aerrors.HandleExternalError(eerr, "loading proving buckets amt")
+	}
+
+	var bucket cid.Cid
+	err := buckets.Get(bucketID, &bucket)
+	switch err.(type) {
+	case *amt.ErrNotFound:
+		return nil, nil // nothing to do
+	case nil:
+	default:
+		return nil, aerrors.HandleExternalError(err, "getting proving bucket")
+	}
+
+	bhamt, err := hamt.LoadNode(vmctx.Context(), vmctx.Ipld(), bucket)
+	if err != nil {
+		return nil, aerrors.HandleExternalError(err, "failed to load proving bucket")
+	}
+
+	err = bhamt.ForEach(vmctx.Context(), func(k string, val interface{}) error {
+		maddr, err := address.NewFromBytes([]byte(k))
+		if err != nil {
+			return aerrors.Escalate(err, "parsing miner address")
+		}
+
+		params, err := SerializeParams(&CheckMinerParams{NetworkPower: self.TotalStorage})
+		if err != nil {
+			return err
+		}
+
+		ret, err := vmctx.Send(maddr, MAMethods.CheckMiner, types.NewInt(0), params)
+		if err != nil {
+			return err
+		}
+
+		if len(ret) == 0 {
+			return nil // miner is fine
+		}
+
+		var power types.BigInt
+		if err := power.UnmarshalCBOR(bytes.NewReader(ret)); err != nil {
+			return xerrors.Errorf("unmarshaling CheckMiner response (%x): %w", ret, err)
+		}
+
+		if power.GreaterThan(types.NewInt(0)) {
```

Comment:
if this is ever not true, we have some non-determinism in the big int encoding (zero should always be encoded as an empty array).

Reply:
That still one byte - cbg.MajByteString